lib/Provider/Command/MessageSend.php

```diff
@@ -0,0 +1,137 @@
+	 *
+	 * @throws SendException on failure, check message for reason
+	 *
+	 */
+	public function perform(string $userId, string $serviceId, IMessage $message, array $options = []): LocalMessage {
+		// find user mail account details
+		try {
+			$account = $this->accountService->find($userId, (int)$serviceId);
+		} catch (\Throwable $e) {
+			throw new SendException('Error: occurred while retrieving mail account details', 0, $e);
+		}
+		// convert mail provider message to mail app message
+		$localMessage = new LocalMessage();
+		$localMessage->setType($localMessage::TYPE_OUTGOING);
+		$localMessage->setAccountId($account->getId());
+		$localMessage->setSubject($message->getSubject());
+		$localMessage->setBody($message->getBody());
+		// disabled due to issues caused by opening these messages in gui
+		//$localMessage->setEditorBody($message->getBody());
+		$localMessage->setHtml(true);
+		$localMessage->setSendAt($this->time->getTime());
+		// convert all mail provider attachments to local attachments
+		$attachments = [];
+		if (count($message->getAttachments()) > 0) {
+			// iterate attachments and save them
+			foreach ($message->getAttachments() as $entry) {
+				// determine if required parameters are set
+				if (empty($entry->getName()) || empty($entry->getType()) || empty($entry->getContents())) {
+					throw new SendException('Invalid Attachment Parameter: MUST contain values for Name, Type and Contents');
+				}
+				// convert mail provider attachment to mail app attachment
+				try {
+					$attachments[] = $this->attachmentService->addFileFromString(
+						$userId,
+						$entry->getName(),
+						$entry->getType(),
+						$entry->getContents()
+					)->jsonSerialize();
+				} catch (\Throwable $e) {
+					throw new SendException('Error: occurred while saving mail message attachment', 0, $e);
+				}
+			}
+		}
+		// determine if required To address is set
+		if (empty($message->getTo()) || empty($message->getTo()[0]->getAddress())) {
+			throw new SendException('Invalid Message Parameter: MUST contain at least one TO address with a valid address');
+		}
+		// convert recipient addresses
+		$to = $this->convertAddressArray($message->getTo());
+		$cc = $this->convertAddressArray($message->getCc());
+		$bcc = $this->convertAddressArray($message->getBcc());
+		// save message for sending
+		try {
+			$localMessage = $this->outboxService->saveMessage(
+				$account,
+				$localMessage,
+				$to,
+				$cc,
+				$bcc,
+				$attachments
+			);
+		} catch (\Throwable $e) {
+			throw new SendException('Error: occurred while saving mail message', 0, $e);
+		}
+		// evaluate if system messages are to be sent instantly
+		if ($this->config->getAppValue(Application::APP_ID, 'send_system_messages_instantly', 'yes') === 'yes') {
+			try {
+				$localMessage = $this->outboxService->sendMessage($localMessage, $account);
+			} catch (\Throwable $e) {
+				throw new SendException('Error: occurred while sending mail message', 0, $e);
+			}
+		}
```

Comment:
This is out of scope. We do not want async sending for now.

Ref https://github.com/nextcloud/mail/pull/9651#issuecomment-2323876268.

This conditional has to go. Always invoke ``sendMessage``.

Reply:
Sure, I can do that. I was just going to make this configurable, and do a PR for the user interface, so that the user can change this option. But I'll remove it.